packaging/helm-charts/ksml/templates/prometheus-rule.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" )  .Values.prometheus.enabled .Values.prometheusRule.enabled }}
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" )  .Values.prometheus.enabled .Values.prometheusRule.enabled (eq "statefulset" (lower .Values.deploymentMode)) }}
```

Comment:
Can we restructure these so user intent (values.yaml) is evaluated first?
So first the enabled check, then the statefulset limit, then the APIVersions limit?

I know this MR didn't introduce it, but now that the statement it's getting bigger I think it can help increase readability.